### ISSUE TITLE
[Request] Option to float screenshots on screen after capture

### ISSUE THREAD

**Issue by favna** [ShareX/ShareX#2363]
Making screenshots just to reference them elsewhere but only temporarily would be greatly improved by allowing screenshots to float in a small window (preferably with "always on top" property) that is also scalable and moveable to anywhere on the screen. An example of this can be seen in the [iOS jailbreak tweak Snapper ](https://youtu.be/49Gjl_Rru4o)

To sketch an example situation in which this would be useful:
Say you are editing a document and you want to have it maximized, but you need to reference a body of text from your browser. In normal circumstances you would have to either copy the text (if at all possible, we all know those pesky sites that block copying) or alt+tab switch between windows a LOT. By having floating screencaptures you can screencap the text from the browser and go to your text editor which very likely has some space somewhere that is not taken up (i.e. In MS Word the side "bars" around the page) where you could position the screencap.

Due to the way this would probably be implemented I can see it being a feature under "after capture task so while for the previous example in Snapper the snap is not instantly saved, for ShareX that would be fully up to the user.

**Comment by campbeb** [comment]
That would be interesting feature.
If you are looking for something that could immediately solve your example situation, look at OnTopReplica (https://ontopreplica.codeplex.com/)

**Comment by Scrxtchy** [comment]
I have a similar setup already but uses a bunch of other software
I have a workflow that just opens the image in Jpegview, my image viewer, and deletes when it's closed
And then there's an autohotkey to keep it on top optionally

but having this streamlined into the application could provide a benefit similar to note taking software

**Comment by Favna** [comment]
Adding to this. I originally referred to Snapper2 (jailbreak tweak) as inspiration for the feature however iOS 11 is getting this type of feature stock as well so it is possible to simply look at the iOS developer beta now.

**Comment by Favna** [comment]
rip in stale. This feature would still be nice to get so hopefully this comment prevents the bot from closing it.

**Comment by Favna** [comment]
Just gonna say it again, but no bot this should not be stale. This feature will always stay requested until added.

**Comment by Favna** [comment]
no

**Comment by Favna** [comment]
bump

**Comment by Favna** [comment]
No bad bot

**Comment by Favna** [comment]
bump

**Comment by Favna** [comment]
No. Bad bot. Keep issue open.

**Comment by Favna** [comment]
How about no

**Comment by oheadman** [comment]
Your feature is like Snipaste software functions, it would be nice if ShareX add it too!

**Comment by honkerjha** [comment]
if  ShareX add the function, it will be the best software!

**Comment by tswwe** [comment]
I agree that this feature is really useful, and the *Snipaste* software you mentioned does it well. Now I am using *ShareX* together with *Snipaste*, though it's better for *ShareX* to include this feature.
FYI, [here](https://docs.snipaste.com/getting-started?id=paste) is the *Snipaste* software's document about this *Paste* feature (I personally prefer naming it a *Pin* feature). This feature basically converts the content (supports plain text, filepath, file, color, html, image, etc.) in the system clipboard into an image and makes it a topmost floating window. The floating window can be moved, zoomed, rotated, copied (as image or back to the origin stuff in the clipboard), exported to an image file, grouped, hided, opacity-tuned, color-picked and more.

**Comment by moniuch** [comment]
This is in no way stale stuff. This is a hot feature to still be added.

Use cases:

- to compare two states/windows by overlaying one on another (with some transparency) or putting side by side with a slight overlap.
- small screen estate setups
- short-lived snapshots that do not clutter your ss folder when done

Make it movable, add custom transparency, just like Notepad++ or Terminal.

**Comment by zhupengfeivip** [comment]
This is a great feature. When will it be available?

**Comment by tonyho** [comment]
Hope this function will be available in the future.

**Comment by EricTetz** [comment]
>  allowing screenshots to float in a small window (preferably with "always on top" property) that is also scalable and moveable to anywhere on the screen [..] you can screencap the text from the browser and go to your text editor which very likely has some space somewhere [..] where you could position the screencap.

ShareX already has this and I use it all the time exactly as described:

![image](https://user-images.githubusercontent.com/929211/154997093-9ebb4e6b-0f85-4ebc-a719-574f158385e3.png)

Set the editor to auto-size.  An "Always on top" flag would be icing, but the cake is there.

**Comment by favna** [comment]
@EricTetz that relies on the image editor which is a bypass if anything. It isn't like the requested feature at all.

**Comment by moniuch** [comment]
@EricTetz We know the cake is there. If you read OP again, the idea is to have the icing (always on top). A floating window (like any of Adobe's tool palettes) would allow to have the screenshot open as a reference, especially valuable when working on small screen estate, eliminating the need of switching apps back and forth.

**Comment by cafernan** [comment]
Hi. I also would like to have this feature in Sharex. I've used it (a lot) on the software https://www.apowersoft.com/free-screen-capture  where you use the "pin" button to activate it. The "open image editor" workaround could help, but it's quite different experience when working, as the "pin" feature, doesn´t show any border or tool bar and it's always on top. Thanks.

**Comment by ProfessorX** [comment]
Same here.

The **pin feature** works fine on Snipaste, it's helpful when I need to screenshot and write docs with reference to that picture
at the same time.
-------

**Comment by Jaex** [comment]
I was thinking of implementing this but it gonna be complicated for people to execute this "Pin to screen" feature. Because unlike other screen capture apps, ShareX don't have region capture window with actions under selection region. So people can't pin the screen from region capture directly. Instead people must rely on using "After capture tasks" which wouldn't be ideal because people not always need to pin the screen. So I was unsure about how to implement this.

**Comment by Jaex** [comment]
Work in progress: https://jaex.getsharex.com/2022/07/GVEuUqmFcS.mp4

**Comment by Jaex** [comment]
Added it as after capture task now:

![](https://jaex.getsharex.com/2022/07/ShareX_qhFxrk2Vue.png)

**Comment by Jaex** [comment]
Also added it to task context menu now:

https://jaex.getsharex.com/2022/07/JE7oB04aBQ.mp4

**Comment by Jaex** [comment]
If you want to test it and give feedback, you can use [dev build](https://getsharex.com/docs/dev-builds).

Current hotkeys:

<kbd>Left Click</kbd> Move image
<kbd>Right Click / Escape</kbd> Close image
<kbd>Middle Click</kbd> Reset image
<kbd>Mouse Wheel</kbd> Image scale
<kbd>Ctrl + Mouse Wheel</kbd> Image opacity
<kbd>Ctrl + C</kbd> Copy image

**Comment by Jaex** [comment]
Added "Pin to screen" tool window:

![](https://jaex.getsharex.com/2022/07/ShareX_vFNhMrRrRu.png)

Updated previous message with all changes.

**Comment by cei8** [comment]
> 

"Pin to screen from clipboard"
"Pin to screen from screen..."
"Pin to screen from file..."

Could you make them a submenu of “Pin to screen"?

I just need "Pin to screen from screen..." that support hotkey.

**Comment by Issam-seghir** [comment]
This future does not exist anymore ?

**Comment by CoenraadS** [comment]
I just tried the dev build and its a great start, but I think it could use a few usability tweaks.

Current it is not intuitive how to interact with the pinned screenshot, and it is also sometime hard to distinguish the pinned item from the screen.

I think something like the how CleanShot X does it could be nice

![image](https://user-images.githubusercontent.com/5593825/208971002-7aef2f20-9ab3-410d-b1c7-f09ab51016b1.png)

Basically on mouse over, you get the following options. And by default its just a small window in the corner, clicking pin will expand it to its true size and perhaps give more options.

**Comment by vassilyzaitsev** [comment]
I can't seem to be able to add a border to it tho... it's kinda weird when I try to do a region pin.. but it doesn't look like it did anything until I click and drag... it is camouflaged lol 

The configs can't seem to be able to add a border to the task "Pin to screen" because it only adds effects after capture.. 

Adding effects to the "Pin to screen" would be nice :)

**Comment by favna** [comment]
I encourage anyone with further requests to create the own issue(s). As you can see, this issue was eventually closed and has recently (January 18th) been released to stable. As always, those who never give up will eventually get their way. Even if that takes re-bumping the issue many times to battle against Stale bot. 

My personal request has now landed in stable so I will be unsubscribing from this issue.

Thank you so much ShareX team for your amazing work on this program, my life would be incomplete without it.

**Comment by Jaex** [comment]
> I can't seem to be able to add a border to it tho... it's kinda weird when I try to do a region pin.. but it doesn't look like it did anything until I click and drag... it is camouflaged lol
> 
> The configs can't seem to be able to add a border to the task "Pin to screen" because it only adds effects after capture..
> 
> Adding effects to the "Pin to screen" would be nice :)

![](https://jaex.getsharex.com/2023/10/chrome_Hk1mkdMLwG.png)

**Comment by namoran** [comment]
I use windows 10 built in virtual desktops. The pins are great, thanks for adding. 
Something interesting about them is that they float on top of all desktops, I have only see a few programs that do that, this one, and the detached edge sidebar.
In contrast, the sharex action toolbar is anchored to 1 desktop, whichever one it was opened on. That is a shame as its functionality is useful on every desktop. So I am asking, how did you get the pins to float over all desktops, and can you make them optionally stick to 1 desktop? and also, can you make the action toolbar float over all desktops?
